lazy_theta_star_p_planner/src/lazy_theta_star_p_planner.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <memory>
+#include <string>
+#include "lazy_theta_star_p_planner/lazy_theta_star_p_planner.hpp"
+#include "lazy_theta_star_p_planner/lazy_theta_star_p.hpp"
+
+namespace lazyThetaStarP_Planner
+{
+void LazyThetaStarP_Planner::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmapRos)
+{
+  planner_ = std::make_unique<lazyThetaStarP::LazyThetaStarP>();
+  node_ = parent;
+  name_ = name;
+  tf_ = tf;
+  planner_->costmap_ = costmapRos->getCostmap();
+  globalFrame_ = costmapRos->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+      8));
+
+  node_->get_parameter(name_ + ".how_many_corners", planner_->how_many_corners_);
+
+  if (planner_->how_many_corners_ != 8 && planner_->how_many_corners_ != 4) {
+    planner_->how_many_corners_ = 8;
+    RCLCPP_WARN(node_->get_logger(),
+      "Your value for - .how_many_corners  was overridden, and is now set to 8");
+  }
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".euc_tolerance", rclcpp::ParameterValue(1.5));
+
+  node_->get_parameter(name_ + ".euc_tolerance", planner_->euc_tolerance_);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".costmap_tolerance", rclcpp::ParameterValue(1.0));
```

Comment:
What's costmap tolerance?

Reply:
I have described about it in the .cpp file. It is the weight for the costmap traversal cost.

Reply:
Got it - lets call it traversal cost then.